src/animationComponents/EntityScopes.js

```diff
@@ -498,6 +503,8 @@ const CompoundEntity = React.memo(React.forwardRef(({ id, index, indexArray=[],
         // Get the position of the rigid body, tis is world position
         const pos = flattenedParticleRefs.current[i].current.translation();
         dummy.position.set(pos.x, pos.y, pos.z);
+
+        all_positions.push( new THREE.Vector3( pos.x, pos.y, pos.z ))
```

Comment:
It we create a blob object at the deepest CompoundEntity then we can reuse that set of Particles for finding the convex hull at the next scope. This would reduce the computation. I've been using the ParticleRigidBody userData property to track information (e.g. color) we could have an array that is the same length as config.entityCounts which indicates if the Particle is at the boundary of a blob at a particular scope. So at scope 0 to find the boundary particles we only need to consider particle which are at the boundary of scope 1 blobs. There is a further optimisation if we classify Particles as on the "inside" or "outside" of the CompoundEntity, particles on the inside cannot be at the boundary so we would not need to check them.

I pushed the code I'm working on to a branch https://github.com/markNZed/react-three/tree/scaleAndRelations where you can see how I am dealing with the userData (it got a bit complicated because we need to remember the user data across React re-rendering e..g when the size of the particle changes). https://github.com/markNZed/react-three/blob/45f1f2bf80ed0463e5fcf707c745dee7b983101b/src/animationComponents/EntityScopes.js#L212

It probably makes sense to just find the boundary particles at each CompoundEntity by looping over all particles in that CompoundEntity (flattenedParticleRefs). Then we can merge that code and the optimizations described here could be implemented. Still probably work capturing the ideas in code comments as the text here will get "lost" quickly. Performance may not be a blocking issue anyway so let's see if it need optimization first.